src/transformStateWithClones.js

```diff
@@ -7,7 +7,67 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const properties = {
+    addProperties: 'addProperties',
+    clear: 'clear',
+    removeProperties: 'removeProperties',
+  };
+  const newObj = { ...state };
+  let obj = {};
+  const newArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case properties.addProperties:
```

Comment:
there is no need to create a separate object for action namings. you can use the name directly here
```suggestion
      case 'addProperties':
```